certbot/tests/client_test.py

```diff
@@ -60,15 +60,13 @@ def test_it(self):
                 self._call()
 
     @mock.patch("certbot.account.report_new_account")
-    @mock.patch("certbot.client.display_ops.get_email")
-    def test_email_retry(self, _rep, mock_get_email):
```

Comment:
This is almost ready! Except you should put `test_email_retry` back in the test suite, in addition to the new `test_email_invalid_noninteractive` :).  You'll probably need to explicitly set `self.config.noninteractive_mode = False` to get it to work.